spec/Appendix A -- Field Selection.md

```diff
@@ -0,0 +1,549 @@
+designed to express semantic equivalence between arguments of a field and fields within the result 
+type. Specifically, it allows defining complex relationships between input arguments and fields in
+the output object by encapsulating these relationships within a parsable string format. It is used
+in the `@is` and `@requires` directives.
+
+To illustrate, consider a simple example from a GraphQL schema:
+
+```graphql
+type Query {
+    userById(userId: ID! @is(field: "id")): User! @lookup
+}
+```
+
+In this schema, the `userById` query uses the `@is` directive with {FieldSelectionMap} to declare that
+the `userId` argument is semantically equivalent to the `User.id` field. 
+
+An example query might look like this:
+
+```graphql
+query {
+    userById(userId: "123") {
+        id
+    }
+}
+```
+
+Here, it is exptected that the `userId` "123" corresponds directly to `User.id`, resulting in the
+following response if correctly implemented:
+
+```json
+{
+    "data": {
+        "userById": {
+            "id": "123"
+        }
+    }
+}
+```
+
+The {FieldSelectionMap} scalar type is used to establish semantic equivalence between an argument and
+the fields within the associated return type. To accomplish this, the scalar must define the
+relationship between input fields or arguments and the fields in the resulting object. 
+Consequently, a {FieldSelectionMap} only makes sense in the context of a specific argument and its 
+return type.
+
+The {FieldSelectionMap} scalar is represented as a string that, when parsed, produces a {SelectedValue}.
+
+A {SelectedValue} must exactly match the shape of the argument value to be considered
+valid. For non-scalar arguments, you must specify each field of the input type in
+{SelectedObjectValue}.
+
+```graphql example
+extend type Query {
+    findUserByName(user: UserInput! @is(field: "{ firstName: firstName }")): User @lookup
+}
+```
+
+```graphql counter-example
+extend type Query {
+    findUserByName(user: UserInput! @is(field: "firstName")): User @lookup
+}
+```
+
+
+## Language
+
+According to the GraphQL specification, an argument is a key-value pair in which the key is the name
+of the argument and the value is a `Value`.
+
+The `Value` of an argument can take various forms: it might be a scalar value (such as `Int`,
+`Float`, `String`, `Boolean`, `Null`, or `Enum`), a list (`ListValue`), an input object
+(`ObjectValue`), or a `Variable`.
+
+Within the scope of the {FieldSelectionMap}, the relationship between input and output is
+established by defining the `Value` of the argument as a selection of fields from the output object.
+
+Yet only certain types of `Value` have a semantic meaning. 
+`ObjectValue` and `ListValue` are used to define the structure of the value. 
+Scalar values, on the other hand, do not carry semantic importance in this context, and variables
+are excluded as they do not exist. 
```

Comment:
I wouldn't say variables don't exist, but we may consider them out of scope for now. I think there are legitimate use cases for allowing them though.